### ISSUE TITLE
[no-throw-literal] Allow DOMException to be thrown in browser env

### ISSUE THREAD

**Issue by PaperStrike** [typescript-eslint/typescript-eslint#3712]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "env": {
    "browser": true
  },
  "rules": {
    "@typescript-eslint/no-throw-literal": ["error"]
  }
}
```

```TS
throw new DOMException('Aborted', 'AbortError');
```

**Expected Result**

No error.

**Actual Result**

Error, expected an error object to be thrown.

**Additional Info**

This seems more like a design problem. ESLint passes as their limitations of static analysis. ESLint isn't about to change the designed behaviour now. https://github.com/eslint/eslint/issues/14880

> Any web platform API using promises to represent operations that can be aborted must adhere to the following:
> * ...
> * Convey that the operation got aborted by rejecting the promise with an **"`AbortError`" `DOMException`**.
> * ...
>
> —— [Using `AbortController` and `AbortSignal` objects in APIs | DOM Standard](https://dom.spec.whatwg.org/#abortcontroller-api-integration)

It is not always a good practice to only throw `Error` in browser environment, as it is encouraged to write a function like:

```js
const abortableFunc = async ({ signal }) => {
  if (signal.aborted) {
    throw new DOMException('Aborted', 'AbortError');
  }
  // ...
}
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.29.0` |
| `@typescript-eslint/parser`        | `4.29.0` |
| `TypeScript`                       | `4.3.5` |
| `ESLint`                           | `7.32.0` |
| `node`                             | `14.15.0` |

**Comment by bradzacher** [comment]
It's an oversight of sorts because `DOMException` is not declared as an extension of the `Error` base class.
https://github.com/microsoft/TypeScript/blob/c0796c1dfb62198976474c09350e98a1227d17fb/lib/lib.dom.d.ts#L3708-L3737

I think it's just a very, very, very rare usecase to use `DOMException` yourself, so it's not a case really anyone runs into.

**Comment by PaperStrike** [comment]
Yes, it's rare to use currently. Well, DOM Standard gives an example that uses DOMException, in https://dom.spec.whatwg.org/#aborting-ongoing-activities .
The standard used the reject executer, and I prefer to use throw in async functions, which has the same functionality.

I bet it will be widely used by browser library developers in a near future. :)

**Comment by PaperStrike** [comment]
Ta-da! `DOMException` now extends `Error`.
